src/views/Detail.vue

```diff
@@ -131,58 +143,69 @@ export default {
+          return DirectoryDetail;
+        case Types.video:
+          return VideoDetail;
+        case Types.audio:
+          return AudioDetail;
+        default:
+          return null;
+      }
     },
-    carouselIndex: {
-      get() {
-        return this.items.findIndex((item) => item.hash === this.fileHash);
-      },
-      set(index) {
-        // TODO: Fix issues with duplicate results
-
-        // TODO: Change page_size to pageSize in store.
-        // eslint-disable-next-line camelcase
-        const { page_size } = this.$store.state.results[this.fileType].results;
-
-        // eslint-disable-next-line camelcase
-        const page = this.primaryPage + Math.floor(index / page_size);
-
-        this.$router.replace({
-          ...this.$route,
-          query: {
-            ...this.$route.query,
-            ...page,
-          },
-          params: {
-            ...this.$route.params,
-            fileHash: this.items[index].hash,
-          },
-        });
-      },
-    },
-    detailType() {
-      return this.$store.state.query.type;
+    items() {
+      return store.state.results?.[this.fileType]?.results?.hits;
     },
   },
   watch: {
+    /**
+     * handle previous/next page loading
+     */
     carouselIndex: {
-      handler(index) {
-        const page = Number(this.$route.query.page);
-        if (index === this.items.length - 1) {
-          console.debug('last item: loading items for page', page + 1);
-          this.$store.dispatch(`results/${this.$props.fileType}/getResults`, page + 1);
-        } else if (index === 0 && page > 1) {
-          console.debug('first item: loading items for page', page - 1);
-          this.$store.dispatch(`results/${this.$props.fileType}/getResults`, {
-            page: page - 1,
-            prepend: true,
-          }).then(() => {
-            this.primaryPage -= 1;
+      handler(index, previousIndex) {
+        // update the fileHash and the page number in the url
+        if (previousIndex) {
+          this.$router.replace({
+            ...this.$route,
+            query: {
+              ...this.$route.query,
+              page: 1 + Math.floor(index / pageSize),
+            },
+            params: {
+              ...this.$route.params,
+              fileHash: this.items[index].hash,
+            },
           });
         }
+
+        // handle fetching missing items from the api
+        const page = Number(this.$route.query.page);
+        if (index === this.items.length - 1
+          || (index < this.items.length - 1 && this.items[index + 1] === undefined)) {
+          console.debug('last page item: loading items for page', page + 1);
+          store.dispatch(`results/${this.fileType}/fetchPage`, { page });
+        } else if (index === ((page - 1) * 15) && page > 1) {
+          console.debug('first page item: loading items for page', page - 1);
+          store.dispatch(`results/${this.fileType}/fetchPage`, { page: page - 2 });
+        }
```

Comment:
The `+1` thing is a real issue, also for maintenance. I would suggest using a single place (e.g. property) where explicitly transform between 0 and 1-indexed pages. I would suggest using 0-indexed pages on the store (and API) side and 1-indexed pages on the view-side.

Reply:
I believe that is exactly what I did. Maybe it would help for readability to for example call 0-indexed pages 'batches' in stead.

Reply:
ok i did that, it is more consistent now.